src/functions.js

```diff
@@ -0,0 +1,32 @@
+const tasks = JSON.parse(localStorage.getItem('tasks')) || [];
+
+function saveTasks() {
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+}
+
+function addTask(description) {
+  const newTask = {
+    description,
+    completed: false,
+    index: tasks.length + 1,
+  };
+  tasks.push(newTask);
+  saveTasks();
+}
+
+function deleteTask(index) {
+  tasks.splice(index, 1);
+  tasks.forEach((task, i) => {
+    task.index = i;
+  });
+  saveTasks();
+}
```

Comment:
- [ ] kindly make sure that if you delete the task it updates all remaining items' indexes, so they represent the current list order and are unique. to meet the below requirement
-> Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique(i.e. if you're deleting the first task index 1 from the list, the index of the next task(2) should be set to 1).. 
be because currently when I delete a task it just deletes that task but the index is not being updated 
<img width="1438" alt="Screen Shot 2023-06-24 at 22 59 23" src="https://github.com/ntimanu/To-do-list/assets/60146030/dfff903d-00bd-4396-af26-cf405707bb28">

Reply:
Done